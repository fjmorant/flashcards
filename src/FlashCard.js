/* @flow */

import {observable, action} from 'mobx'
import shortid from 'shortid'

export default class FlashCard {
    id = shortid.generate()

    @observable name = ''
    @observable meaning = ''
    @observable example = ''
    @observable mastered = false

    name: string
    meaning: string
    example: string
    mastered: boolean

    @action.bound
    changeName(name: string) {
        this.name = name
    }

    @action.bound
    changeMeaning(meaning: string) {
        this.meaning = meaning
    }

    @action.bound
    changeExample(example: string) {
        this.example = example
    }

    @action.bound
    toggleMaster() {
        this.mastered = !this.mastered
    }
}