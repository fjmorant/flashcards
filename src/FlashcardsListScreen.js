import React, {Component} from 'react'
import {StyleSheet, Text, View, Button, TouchableOpacity} from 'react-native'
import List from './common/List'
import Swipeout from 'react-native-swipeout'
import {observer, inject} from 'mobx-react'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgb(245, 252, 255)',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: 'rgb(51, 51, 51)',
    marginBottom: 5,
  },
})

@inject('flashCardList')
@observer
class FlashcardsListScreen extends Component {
  static navigationOptions = ({navigation, screenProps}) => {
    return {
      title: 'Flash Cards',
      headerRight: (
        <Button title={'Add'} onPress={() => navigation.navigate('Add')} />
      ),
      headerLeft: (
        <Button
          title={'Practise'}
          onPress={() => navigation.navigate('View')}
        />
      ),
    }
  }

  renderFlashCard = flashcard => {
    const swipeoutBtns = [
      {
        text: 'Delete',
        backgroundColor: 'red',
        onPress: () => this.props.flashCardList.delete(flashcard.id),
      },
    ]

    return (
      <Swipeout autoClose right={swipeoutBtns}>
        <TouchableOpacity
          onPress={() =>
            this.props.navigation.navigate('Add', {id: flashcard.id})
          }>
          <View style={{padding: 8}}>
            <Text style={{color: 'rgb(0,0,0)'}}>Name : {flashcard.name}</Text>
            <Text style={{color: 'rgb(0,0,0)'}}>
              Meaning : {flashcard.meaning}
            </Text>
            <Text style={{color: 'rgb(0,0,0)'}}>
              Example : {flashcard.example}
            </Text>
            <Text style={{color: 'rgb(0,0,0)'}}>
              Mastered : {(flashcard.mastered || false).toString()}
            </Text>
          </View>
        </TouchableOpacity>
      </Swipeout>
    )
  }

  render() {
    return (
      <View>
        <List
          items={this.props.flashCardList.list}
          renderItem={this.renderFlashCard}
        />
      </View>
    )
  }
}

export default FlashcardsListScreen
