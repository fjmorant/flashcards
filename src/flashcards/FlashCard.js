/* @flow */

import {observable, action} from 'mobx'

export default class FlashCard {
    @observable name = ''
    @observable meaning = ''
    @observable example = ''
    @observable mastered = false

    name: string
    meaning: string
    example: string
    mastered: boolean

    @action
    changeName(name) {
        this.name = name
    }

    @action
    changeMeaning(meaning) {
        this.meaning = meaning
    }

    @action
    changeExample(example) {
        this.example = example
    }

    @action
    toggleMaster() {
        this.mastered = !this.mastered
    }
}