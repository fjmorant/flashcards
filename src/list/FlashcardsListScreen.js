import React, {Component} from 'react';
import {
  StyleSheet,
  Text,
  View,
  Button,
  TouchableOpacity,
} from 'react-native';
import List from '../common/List';
import immutable from 'immutable';
import {connect} from 'react-redux';
import {loadFlashCardsFromPersistance, deleteFlashCard} from './flashcardsDuck'
import Swipeout from 'react-native-swipeout'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgb(245, 252, 255)',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: 'rgb(51, 51, 51)',
    marginBottom: 5,
  },
})

class FlashCards extends Component {

  constructor(props) {
    super(props)
    this.renderFlashCard = this.renderFlashCard.bind(this)
  }

  componentDidMount() {
    this.props.loadFlashCardsFromPersistance();
  }

  static navigationOptions = {
    title: 'Flash Cards',
    header: ({navigate, state}) => {
      let right = (
        <Button
          title={'Add'}
          onPress={() => navigate('Add')}
        />
      )

      let left = (
        <Button
          title={'Practise'}
          onPress={() => navigate('View')}
        />
      )

      return {
        right,
        left,
      }
    },
  }

  renderFlashCard(flashcard) {
    const swipeoutBtns = [
      {
        text: 'Delete',
        backgroundColor: 'red',
        onPress: () => this.props.deleteFlashCard(flashcard.get('id')),
      }
    ]

    return (
      <Swipeout
          autoClose
          right={swipeoutBtns}>
          <TouchableOpacity onPress={() => this.props.navigation.navigate('Add', {id: flashcard.get('id')})}>
            <View style={{padding: 8}}>
              <Text style={{color: 'rgb(0,0,0)'}}>
                Name : {flashcard.get('name')}
              </Text>
              <Text style={{color: 'rgb(0,0,0)'}}>
                Meaning : {flashcard.get('meaning')}
              </Text>
              <Text style={{color: 'rgb(0,0,0)'}}>
                Example : {flashcard.get('example')}
              </Text>
            </View>
          </TouchableOpacity>
      </Swipeout>
    );
  }

  render() {
    return (
      <View>
          <List
            items={this.props.flashcards}
            renderItem={this.renderFlashCard}/>
      </View>
    )
  }
}

export default connect(state => ({
    flashcards: state.flashcards.get('flashcards') ? state.flashcards.get('flashcards').reverse() : immutable.List(),
  }),
  (dispatch) => ({
    loadFlashCardsFromPersistance: () => dispatch(loadFlashCardsFromPersistance()),
    deleteFlashCard: (id) => dispatch(deleteFlashCard(id)),
  }),
)(FlashCards);
