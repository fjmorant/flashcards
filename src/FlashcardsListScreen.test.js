import 'react-native'
import React from 'react'
import FlashcardsListScreen from './FlashcardsListScreen'
import FlashCard from './FlashCard'

import renderer from 'react-test-renderer'

const flashCardList = {
  list: [new FlashCard()],
}

describe('FlashcardsListScreen', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<FlashcardsListScreen flashCardList={flashCardList} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
