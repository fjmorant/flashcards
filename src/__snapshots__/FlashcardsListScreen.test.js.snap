// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashcardsListScreen renders correctly 1`] = `
<View>
  <RCTScrollView
    dataSource={
      ListViewDataSource {
        "items": 1,
      }
    }
    enableEmptySections={true}
    renderRow={[Function]}
    renderScrollComponent={[Function]}
    renderSeparator={[Function]}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#dbddde",
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          onLayout={[Function]}
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Array [
              Object {},
              Object {
                "left": 0,
              },
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "padding": 8,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "rgb(0,0,0)",
                  }
                }
              >
                Name : 
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "rgb(0,0,0)",
                  }
                }
              >
                Meaning : 
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "rgb(0,0,0)",
                  }
                }
              >
                Example : 
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "rgb(0,0,0)",
                  }
                }
              >
                Mastered : 
                false
              </Text>
            </View>
          </View>
        </View>
        <View />
        <View />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
